tests/unit/test_coverage.py

```diff
@@ -114,3 +114,119 @@ def test_generate_coverage_markdown(mocker):
+                num_changed_lines=2,
+                files={
+                    pathlib.Path("codebase/code.py"): coverage.FileDiffCoverage(
+                        path=pathlib.Path("codebase/code.py"),
+                        percent_covered=decimal.Decimal("0.5"),
+                        violation_lines=[3],
+                    )
+                },
+            ),
+        ),
+        (
+            {pathlib.Path("codebase/code2.py"): [1, 3]},
+            [1, 2],
+            [3],
+            coverage.DiffCoverage(
+                total_num_lines=0,
+                total_num_violations=0,
+                total_percent_covered=decimal.Decimal("1"),
+                num_changed_lines=2,
+                files={},
+            ),
+        ),
+        (
+            {pathlib.Path("codebase/code.py"): [4, 5, 6]},
+            [1, 2, 3],
+            [7],
+            coverage.DiffCoverage(
+                total_num_lines=0,
+                total_num_violations=0,
+                total_percent_covered=decimal.Decimal("1"),
+                num_changed_lines=3,
+                files={
+                    pathlib.Path("codebase/code.py"): coverage.FileDiffCoverage(
+                        path=pathlib.Path("codebase/code.py"),
+                        percent_covered=decimal.Decimal("1"),
+                        violation_lines=[],
+                    )
+                },
+            ),
+        ),
+    ],
+)
+def test_get_diff_coverage_info(
+    coverage_obj_no_branch, added_lines, executed_lines, missing_lines, expected
+):
+    cov_file = coverage_obj_no_branch.files[pathlib.Path("codebase/code.py")]
+    cov_file.executed_lines = executed_lines
+    cov_file.missing_lines = missing_lines
+    result = coverage.get_diff_coverage_info(
+        added_lines=added_lines, coverage=coverage_obj_no_branch
+    )
+    assert result == expected
+
+
+def test_get_added_lines(git):
+    diff = (
+        """+++ b/README.md\n@@ -1,2 +1,3 @@\n-# coverage-comment\n+coverage-comment\n"""
+    )
+    git.register("git fetch origin main --depth=1000")()
+    git.register("git diff --unified=0 FETCH_HEAD -- .")(stdout=diff)
+    assert coverage.get_added_lines(git=git, base_ref="main") == {
+        pathlib.Path("README.md"): [1, 2, 3]
+    }
+
+
+@pytest.mark.parametrize(
+    "line_number_diff_line, expected",
+    [
+        ("@@ -1,2 +7,4 @@ foo()", [7, 8, 9, 10]),
+        ("@@ -1,2 +8 @@ foo()", [8]),
+    ],
+)
+def test_parse_line_number_diff_line(git, line_number_diff_line, expected):
+    result = list(coverage.parse_line_number_diff_line(line_number_diff_line))
+    assert result == expected
+
+
+def test_parse_diff_output(git):
+    diff = """diff --git a/README.md b/README.md
+index 1f1d9a4..e69de29 100644
+--- a/README.md
++++ b/README.md
+@@ -1 +1 @@
+-# coverage-comment
+-coverage-comment
+@@ -3,2 +3,4 @@
+-foo
+-bar
++foo1
++bar1
++foo2
++bar2
+--- a/foo.txt
++++ b/foo.txt
+@@ -0,0 +1 @@
++bar
```

Comment:
```suggestion
+bar
@@ -8 +7,0 @@
-foo
```
I think it would also be useful to test for deletions without any additions

Reply:
Good idea !